lib/rules/no-useless-escape.js

```diff
@@ -95,10 +114,12 @@ module.exports = function(context) {
             return;
         }
 
-        (node.raw.match(/\\[^\d]/g) || []).forEach(validate, {
-            "escapes": nodeEscapes,
-            "node": node
-        });
+        while ((match = pattern.exec(node.raw))) {
```

Comment:
@platinumazure this one:

```
{
    "node": node
    "escapes": nodeEscapes,
}
```

I actually think the following would be better and more efficient:

```
var validator = validate.bind({ escapes, nodeEscapes, node: node });
while ((match = pattern.exec(node.raw))) { validator(match); }
```